### PR TITLE
1.Introduction - task for the first lesson of the course.

### PR DESCRIPTION
**Pull request by Yulson**
Prepared tests for demo.redmine.org using TestUnit and Selenium Webdriver.

### CODE REVIEW

**Comment by Yulson** [comment]
@d-konovalov please review

File: Issue.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'faker'
+
+class Issue
+
+	attr_reader :issue_name, :issue_type
+	attr_accessor :issue_name, :issue_type
+
+	def initialize(options={})
+		@issue_name = Faker::Team.name
```
Comment (d-konovalov) [inline Issue.rb]: fix to @name instead of @issue_name

File: Issue.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require 'faker'
+
+class Issue
+
+	attr_reader :issue_name, :issue_type
+	attr_accessor :issue_name, :issue_type
+
+	def initialize(options={})
```
Comment (d-konovalov) [inline Issue.rb]:
```ruby
def initialize(type)
  @name = Faker::Team.name
  @type = type
end
```

File: Main_project_Tests_TestUnit.rb [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  	 	assert_equal(expected_notice, actual_notice)
+   	end
+
+	def test_create_bug
+	    register_user(@user1)
+	    create_new_project(@project1)
+	    create_new_issue(@issue_bug)
+	     	
+	    expected_subject = @issue_bug.issue_name
+	    actual_array_subjects = Array.new
```
Comment (d-konovalov) [inline Main_project_Tests_TestUnit.rb]: var = []

File: Main_project_Tests_TestUnit.rb [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	    @driver.find_elements(:css, '.subject').each {|element| actual_array_subjects.push(element.text)}
```
Comment (d-konovalov) [inline Main_project_Tests_TestUnit.rb]: my_subjects = @driver.find_elements(:css,....).map {|el| el.text}

File: Main_project_Tests_TestUnit.rb [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+	    assert_match(/#{expected_subject}/, actual_array_subjects.to_s)
```
Comment (d-konovalov) [inline Main_project_Tests_TestUnit.rb]: assert_true(my_subjects.include? expected_subject)

**Review by d-konovalov — CHANGES_REQUESTED** [review]

**Comment by Yulson** [comment]
Hi @d-konovalov,
I did all changes according to your comments in main project. Could you please review it again.

File: 4. Operators-and-onstructors/Operators-and-constructors-task1.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
+#1. Create method which gets 10 random integers (from 0 to 9) (generate it using while or until)
```
Comment (d-konovalov) [inline 4. Operators-and-onstructors/Operators-and-constructors-task1.rb:1]: first of all we should have a method as task says
and second: try to reduce lines number as much as possible

File: Main_project_Helpers.rb [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+ 		@driver.find_element(:css, '.icon-add').click
+ 		@driver.find_element(:css, '#project_name').send_keys project.name
+ 		@wait.until {@driver.find_element(:css, '[name="commit"]').displayed?}
+ 		@driver.find_element(:css, '[name="commit"]').click
+ 		project
+ 	end
+
+   def change_roles
+ 		@driver.find_element(:css, '.odd .icon-edit:nth-child(1)').click
+ 		@driver.find_element(:css, '.odd [value="5"]:nth-child(1)').click
```
Comment (d-konovalov) [inline Main_project_Helpers.rb:64]: hardcoded values in locator is a bad sign
if you don't have now ideas how to fix it please ask me for help

**Review by d-konovalov — CHANGES_REQUESTED** [review]

**Pull request merged**